core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerRemoteMeta.java

```diff
@@ -0,0 +1,375 @@
+      super(offset, done, rows);
+      this.statementSlug = statementSlug;
+    }
+
+    /**
+     * Creates a {@code LookerFrame} for the statement slug
+     *
+     * @param statementSlug slug for the prepared statement generated by a Looker instance.
+     * @return the {@code firstFrame} for the result set.
+     */
+    public static final LookerFrame create(String statementSlug) {
+      assert null != statementSlug;
+      return new LookerFrame(0, false, Collections.emptyList(), statementSlug);
+    }
+  }
+
+  LookerSDK getSdk() {
+    if (null == sdk) {
+      sdk = ((LookerRemoteService) service).sdk;
+    }
+    return sdk;
+  }
+
+  /**
+   * Helper to disable any SSL and hostname verification when `verifySSL` is false.
+   */
+  private void trustAllHosts(HttpsURLConnection connection) {
+    // Create a trust manager that does not validate certificate chains
+    TrustManager[] trustAllCerts = new TrustManager[]{
+        new X509TrustManager() {
+          public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+            return null;
+          }
+          public void checkClientTrusted(java.security.cert.X509Certificate[] certs,
+              String authType) {
+          }
+          public void checkServerTrusted(java.security.cert.X509Certificate[] certs,
+              String authType) {
+          }
+        }
+    };
+    // Create all-trusting host name verifier
+    HostnameVerifier trustAllHostNames = (hostname, session) -> true;
+    try {
+      SSLContext sc = SSLContext.getInstance("SSL");
+      sc.init(null, trustAllCerts, new java.security.SecureRandom());
+      connection.setSSLSocketFactory(sc.getSocketFactory());
+      connection.setHostnameVerifier(trustAllHostNames);
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * A regrettable necessity. The Kotlin SDK relies on an outdated Ktor HTTP client based on Kotlin
+   * Coroutines which are a tad difficult to work with in Java. Here we make a HTTP client to handle
+   * the response stream ourselves which is "good enough" for now. This method should be revisited
+   * when the Kotlin SDK has built-in streams.
+   *
+   * TODO https://github.com/looker-open-source/sdk-codegen/issues/1341:
+   *  Add streaming support to the Kotlin SDK.
+   */
+  private void streamResponse(String url, OutputStream outputStream) throws IOException {
+    // use some SDK client helpers to tighten up this call
+    AuthSession authSession = getSdk().getAuthSession();
+    Transport sdkTransport = authSession.getTransport();
+    // makes a proper URL from the API endpoint path as the SDK would.
+    String endpoint = sdkTransport.makeUrl(url, Collections.emptyMap(), null);
+    URL httpsUrl = new URL(endpoint);
+    HttpsURLConnection connection = (HttpsURLConnection) httpsUrl.openConnection();
```

Comment:
You probably want to set some timeouts here, or transport them from configuration somehow. What is the latency for connect, for reads from Looker, and so on. Not really expected to have actual values in code this early but it needs coordination for reasonable values on both sides, and if read latency is allowed to be high then you probably want more buffering.

Reply:
Ah of course, we set a timeout for the SDK Transport that should also be applied here. Thank you for catching this.

Reply:
Added in https://github.com/looker-open-source/calcite-avatica/pull/2/commits/b407594dc18df89006383cd8abedce40d761a336